src/ngraph/frontend/onnx_import/utils/convpool.cpp

```diff
@@ -82,8 +82,10 @@ namespace ngraph
                     Shape pad_shape;
                     for (std::size_t idx = 0; idx < input.size(); ++idx)
                     {
-                        pad_shape.emplace_back((output.at(idx) - 1) * strides.at(idx) +
-                                               kernel.at(idx) - input.at(idx));
+                        pad_shape.emplace_back(
+                            std::max(static_cast<long>((output.at(idx) - 1) * strides.at(idx) +
+                                                       kernel.at(idx) - input.at(idx)),
```

Comment:
@tsocha can you please give me some more info what happens if their is no casting, is it something to do with integer promotion?

Reply:
yes, because without casting we can get a negative value when type is size_t.
Then std::max compare overflowed size_t to 0 and result is overflowed > 0.